firmware/test/mocks/Particle.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <functional>
+#include "../../inc/spark_wiring_string.h"
+
+enum PublishFlag
+{
+    PUBLIC,
+    PRIVATE,
+    NO_ACK,
+    WITH_ACK
+};
+
+struct os_queue_t{
+
+}; 
+
+class MockParticle
+{
+public:
+    MockParticle()
+    {
+    }
+
+public:
+    bool connected() const
+    {
+        return true;
+    }
+
+    bool publish(char const* const szEventName,
+                 char const* const szData,
+                 int /*PublishFlag*/ const flags)
```

Comment:
What is this `/*PublishFlag*/` comment about?

Reply:
it's one of the custom types that Particle defines that I ended up not mocking anywhere, it doesn't need to be there.